torch/utils/_sympy/value_ranges.py

```diff
@@ -500,6 +500,53 @@ def or_(a, b):
     def and_(a, b):
         return ValueRanges.coordinatewise_increasing_map(a, b, sympy.And)
 
+    @staticmethod
+    def _bool_to_int(x):
+        if x.is_singleton():
+            return ValueRanges.wrap(sympy.Integer(1 if x.lower else 0))
+        else:
+            return ValueRanges(sympy.Integer(0), sympy.Integer(1))
+
+    @staticmethod
+    def _bit_length(n):
+        return len(bin(n).lstrip("-0b"))
```

Comment:
lol wtf is this. Do better

Reply:
I took this from https://docs.python.org/3/library/stdtypes.html#int.bit_length because sympy Integer doesn't implement bit_length. Otherwise, I need to write a loop, or use log2.

Reply:
What's the input type here? If it's an int you can just use `bit_length` method on it?

Reply:
It could be sympy Integer, which doesn't have bit_length.

Reply:
If it is guaranteed to be sympy.Integer you can convert it to int